scrapinghub/client/exceptions.py

```diff
@@ -69,6 +73,8 @@ def wrapped(*args, **kwargs):
                 raise ValueTooLarge(http_error=exc)
             elif 400 <= status_code < 500:
                 raise ScrapinghubAPIError(http_error=exc)
+            elif 500 <= status_code < 600:
+                raise ServerError(http_error=exc)
```

Comment:
@chekunkov Reworked a bit: the simplest way would be to call `raise_for_status` in the legacy client, and handle it here, does it look better?